### PR TITLE
Add getUser and setCustomUserClaims mocks

### PR DESCRIPTION
**Pull request by AverageHelper**
# Description
Added mocks for `auth().getUser` and `auth().setCustomUserClaims`.

## Related issues
This helps my own organization test our use of these functions.

## How to test
```
jest
```

### CODE REVIEW

File: mocks/auth.js [hunk 1]
```diff
@@ -18,7 +21,7 @@ class FakeAuth {
 
   deleteUser() {
     mockDeleteUser(...arguments);
-    return Promise.resolve('👍');
+    return Promise.resolve('Success');
```
Comment (sbatson5) [inline mocks/auth.js:24]: lol don't like my emojis? 😂

File: mocks/auth.js [hunk 2]
```diff
@@ -35,6 +38,15 @@ class FakeAuth {
     return Promise.resolve(this.currentUserRecord);
   }
 
+  getUser() {
+    return Promise.resolve(mockGetUser(...arguments));
```
Comment (sbatson5) [inline mocks/auth.js:42]: I like this pattern, which [allows you to do `mockGetUser.mockReturnValue`](https://github.com/Upstatement/firestore-jest-mock/pull/36/files#diff-445ddbc16541e58a306f94edf4461b34R122)

However, if a user doesn't do that, but has an assertion that relies on it being an object, would a test pass?  Would it work if we did:
`return Promise.resolve(mockGetUser(...arguments) || {})`?

File: mocks/auth.js [hunk 3]
```diff
@@ -35,6 +38,15 @@ class FakeAuth {
...
+  }
+
+  setCustomUserClaims() {
+    mockSetCustomUserClaims(...arguments);
+    return Promise.resolve({});
```
Comment (sbatson5) [inline mocks/auth.js:47]: Similar to `getUser` could we have this return `mockSetCustomUserClaims(...arguments)` or an empty object, that way a tester could do `mockSetCustomUserClaims.mockReturnValue` in their own test if they needed to explicitly stub something?

File: mocks/auth.js (hunk 1 above)
Comment (AverageHelper) [reply mocks/auth.js:24]: I love them! But my IDE got confused over the characters 😅
Feel free to revert this if you'd like

File: mocks/auth.js (hunk 2 above)
Comment (AverageHelper) [reply mocks/auth.js:42]: It should. I think I do that in other places but neglected that here.

**Pull request merged**